services/scrape-test/test_samples.py

```diff
@@ -0,0 +1,221 @@
+import datetime
+from google.cloud import bigquery
+from playwright.sync_api import sync_playwright
+import re
+from decimal import Decimal  # If Decimal is used elsewhere
+
+
+def get_rows_from_bq():
+    # Initialize the BigQuery client
+    client = bigquery.Client()
+
+    # Query to retrieve rows from the successful scrapes table
+    query = """
+        SELECT * FROM `annular-net-436607-t0.dbt.testConsumption` LIMIT 500
+    """
+
+    # Run the query and fetch results
+    query_job = client.query(query)
+    results = query_job.result()
+
+    # Convert results to a list of dictionaries for easier handling
+    rows = [dict(row) for row in results]
+    return rows
+
+
+def serialize_row(row):
+    for key, value in row.items():
+        if isinstance(value, (datetime.date, datetime.datetime)):
+            row[key] = value.isoformat()
+        elif isinstance(value, Decimal):
+            row[key] = float(value)
+        elif value is None:
+            row[key] = None  # BigQuery accepts NULL values
+        elif isinstance(value, bool):
+            row[key] = bool(value)
+        else:
+            # Convert other types to strings if necessary
+            row[key] = str(value) if not isinstance(value, (int, float, str)) else value
+    return row
+
+
+def insert_row_to_bq(row, destination_table):
+    client = bigquery.Client()
+
+    # Serialize the row to handle date fields
+    row = serialize_row(row)
+
+    # For debugging: print the row being inserted
+    print(f"Inserting row into {destination_table}: {row}")
+
+    # Insert data into BigQuery
+    errors = client.insert_rows_json(destination_table, [row])  # Wrap row in a list
+    if errors:
+        print(f"Errors occurred while inserting row into {destination_table}: {errors}")
+    else:
+        print(f"Row inserted into {destination_table} successfully.")
+
+
+def scrape_youtube_link(url, context):
+    page = context.new_page()
+
+    try:
+        # Navigate to the target URL
+        page.goto(url, timeout=20000)  # Timeout after 20 seconds if page doesn't load
+
+        # Wait for the page to load completely
+        page.wait_for_load_state("networkidle", timeout=20000)
+
+        # Function to recursively search frames
+        def find_youtube_in_frames(frames):
+            for frame in frames:
+                # Check the frame's URL
+                if "youtube.com" in frame.url:
+                    print(f"Found YouTube iframe with src: {frame.url}")
+                    return frame.url
+
+                # Search for iframes within the frame
+                iframe_elements = frame.query_selector_all("iframe")
+                for iframe in iframe_elements:
+                    child_frame = iframe.content_frame()
+                    if child_frame:
+                        result = find_youtube_in_frames([child_frame])
+                        if result:
+                            return result
+            return None
```

Comment:
Why it's repeated code from dev?

Reply:
This file was only a test script and not meant to be used later on, it's now deleted.